sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -212,6 +224,27 @@ TrackerT invokeNewTracker(RestrictionT restriction) {
         DoFnSignatures.INSTANCE.getSignature((Class) fn.getClass()), fn);
   }
 
+  /**
+   * Internal base class for generated {@link DoFnInvoker} instances.
+   *
+   * <p>This class should <i>not</i> be extended directly, or by Beam users. It must be public for
+   * generated instances to have adequate access, as they are generated "inside" the invoked {@link
+   * DoFn} class.
+   */
+  public abstract static class DoFnInvokerBase<InputT, OutputT, DoFnT extends DoFn<InputT, OutputT>>
```

Comment:
In order to write "normal" Java code to access and mutate the map `onTimerInvokers`, I made this base class. See the javadoc here and elsewhere about visibility. IMO it seems a small price to pay for some readable code and more-or-less foolproof bytebuddy usage.